docs/docs/agents/memory.md

```diff
@@ -0,0 +1,119 @@
+# Memory
+
+There are two types of memory that are relevant when working with agents:
+
+- short-term memory — memory of all of the interactions (e.g., messages) in the same conversation or session (**thread**). This memory is used to enable multi-turn conversations with an agent.
```

Comment:
Consider removing excessive clarifications etc -- so we can fit the concept into a single sentence that does not contain jargon


```suggestion
- short-term memory - allows an agent to remember the current conversation across multiple turns.
- long-term memory - allows an agent to reference information across different conversations.

Note:

shor-term memory is sometimes referred to as thread or session or per thread 
long-term memory is sometimes referred to as **cross-thread** or cross-session memory.